contrib/src/test/java/com/datatorrent/contrib/avro/AvroFileToPojoModuleTest.java

```diff
@@ -0,0 +1,260 @@
+    }
+  }
+
+  @Rule
+  public TestMeta testMeta = new TestMeta();
+
+  private void createAvroInput(int cnt)
+  {
+    recordList = Lists.newArrayList();
+
+    while (cnt > 0) {
+      GenericRecord rec = new GenericData.Record(new Schema.Parser().parse(AVRO_SCHEMA));
+      rec.put("orderId", cnt * 1L);
+      rec.put("customerId", cnt * 2);
+      rec.put("total", cnt * 1.5);
+      rec.put("customerName", "*" + cnt + "*");
+      cnt--;
+      recordList.add(rec);
+    }
+  }
+
+  private void writeAvroFile(File outputFile)
+  {
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(new Schema.Parser().parse(AVRO_SCHEMA));
+    try (DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter)) {
+      dataFileWriter.create(new Schema.Parser().parse(AVRO_SCHEMA), outputFile);
+
+      for (GenericRecord record : recordList) {
+        dataFileWriter.append(record);
+      }
+      FileUtils.moveFileToDirectory(new File(outputFile.getAbsolutePath()), new File(testMeta.dir), true);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Test
+  public void testAvroToPojoModule() throws Exception
+  {
+    try {
+      FileContext.getLocalFSFileContext().delete(new Path(new File(testMeta.dir).getAbsolutePath()), true);
+      int cnt = 7;
+      createAvroInput(cnt);
+      writeAvroFile(new File(FILENAME));
+      createAvroInput(cnt - 2);
+      writeAvroFile(new File(OTHER_FILE));
+
+      avroFileToPojoModule.setAvroFileDirectory(testMeta.dir);
+      avroFileToPojoModule.setPojoClass(SimpleOrder.class);
+
+      AvroToPojo avroToPojo = new AvroToPojo();
+      avroToPojo.setPojoClass(SimpleOrder.class);
+
+      EmbeddedAppLauncherImpl lma = new EmbeddedAppLauncherImpl();
+      Configuration conf = new Configuration(false);
+
+      AvroToPojoApplication avroToPojoApplication = new AvroToPojoApplication();
+      avroToPojoApplication.setAvroFileToPojoModule(avroFileToPojoModule);
+
+      lma.prepareDAG(avroToPojoApplication, conf);
+      EmbeddedAppLauncherImpl.Controller lc = lma.getController();
+      lc.run(10000);// runs for 10 seconds and quits
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  public static class AvroToPojoApplication implements StreamingApplication
+  {
+    AvroFileToPojoModule avroFileToPojoModule;
+
+    public AvroFileToPojoModule getAvroFileToPojoModule()
+    {
+      return avroFileToPojoModule;
+    }
+
+    public void setAvroFileToPojoModule(AvroFileToPojoModule avroFileToPojoModule)
+    {
+      this.avroFileToPojoModule = avroFileToPojoModule;
+    }
+
+    @Override
+    public void populateDAG(DAG dag, Configuration conf)
+    {
+      AvroFileToPojoModule avroFileToPojoModule = dag.addModule("avroFileToPojoModule", getAvroFileToPojoModule());
+      ConsoleOutputOperator consoleOutput = dag.addOperator("console", new ConsoleOutputOperator());
```

Comment:
Since the application is written for the test only, there is no need to use a console operator. You can specify a collector that allows you to verify the results instead. Or, adopt the approach discussed on #660